src/journal.rs

```diff
@@ -39,59 +39,50 @@ enum SyslogLevel {
     Debug = 7,
 }
 
-/// Send a `log::LogRecord` to systemd-journald.
-pub fn log_record(record: &LogRecord) {
+/// Send a `log::Record` to systemd-journald.
+pub fn log_record(record: &Record) {
     let lvl = match record.level() {
-        LogLevel::Error => SyslogLevel::Err,
-        LogLevel::Warn => SyslogLevel::Warning,
-        LogLevel::Info => SyslogLevel::Info,
-        LogLevel::Debug |
-        LogLevel::Trace => SyslogLevel::Debug,
+        Level::Error => SyslogLevel::Err,
+        Level::Warn => SyslogLevel::Warning,
+        Level::Info => SyslogLevel::Info,
+        Level::Debug |
+        Level::Trace => SyslogLevel::Debug,
     } as usize;
-    log_target(lvl, record.location(), record.args(), record.target());
-}
 
-/// Record a log entry, with custom priority and location.
-pub fn log(level: usize, loc: &LogLocation, args: &fmt::Arguments) {
-    send(&[&format!("PRIORITY={}", level),
-           &format!("MESSAGE={}", args),
-           &format!("CODE_LINE={}", loc.line()),
-           &format!("CODE_FILE={}", loc.file()),
-           &format!("CODE_FUNCTION={}", loc.module_path())]);
-}
+    let mut keys = vec![
+        format!("PRIORITY={}", lvl),
+        format!("MESSAGE={}", record.args()),
+        format!("TARGET={}", record.target()),
+    ];
 
-/// Record a log entry, with custom priority, location and target.
-pub fn log_target(level: usize, loc: &LogLocation, args: &fmt::Arguments, target: &str) {
```

Comment:
With the removal of `LogLocation` for embedding directly into `Record`, these functions got unwieldy. If I should put them back, let me know.